services/geolocation/maxmind.go

```diff
@@ -0,0 +1,49 @@
+package geolocation
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"net"
+
+	"github.com/oschwald/maxminddb-golang"
+)
+
+var (
+	ErrInvalidDatabase = errors.New("invalid database file")
+	ErrInvalidIP       = errors.New("ip for lookup cannot be empty or invalid")
+)
+
+type dbReader struct {
+	reader *maxminddb.Reader
+}
+
+func NewMaxmindDBReader(dbLoc string) (*dbReader, error) {
+	reader, err := maxminddb.Open(dbLoc)
+	if err != nil {
+
+		if _, ok := err.(*fs.PathError); ok {
+			return nil, ErrInvalidDatabase
```

Comment:
We can easily add this code path as well in test?

Reply:
We have that in test.